app/models/deal.rb

```diff
@@ -44,6 +44,22 @@ def unpublish
     update(published_at: nil)
   end
 
+  def published?
+    published_at.present?
+  end
+
+  def self.search(search)
```

Comment:
have you read about scopes? if yes then pls explain why you haven't used them here